lib/active_merchant/billing/gateways/safe_charge.rb

```diff
@@ -22,7 +22,8 @@ def initialize(options={})
 
       def purchase(money, payment, options={})
         post = {}
-        add_transaction_data("Sale", post, money, options)
+        post[:sg_APIType] = 1
+        add_transaction_data("Sale3D", post, money, options)
```

Comment:
I think we may want to have the option to flag txns to go through their 3DS system or not at all. It looks like this always defaults to attempting 3DS?

Reply:
Also, Authorizations should be available for attempting 3DS in addition to Purchases, if that is possible on the gateway end.

Reply:
@curiousepic I think I mentioned their internal assessment: if the merchant is not approved for 3DS, Safe Charge will default to `Sale`. And they do not allow for 3DS authorizations, only purchases.

Reply:
Chatted offline about this and it looks like this is the way to go, since the determination of "3DS flow or not" really occurs at the merchant account level. 👍